### PR TITLE
fix: fix rules for sortability

### PR DESCRIPTION
**Pull request by JeoungSulMo**
## To Reviewers
- [ ] Skipping reviews is not a problem

## Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
  - [x] These changes require a usage change
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

## Description

1. The addition of the previous multiple-header function caused a bug in the sorting function. ([code](https://github.com/spaceone-dev/spaceone-design-system/commit/f94fe82ec3bd4e9801774cc8b9739f7b58a5496b#diff-9b42ea78fca9bff6c1aeaf94f636adfae9bae863d7ea6ee05d5ba41ef4884c84R176))
2. The default value of field sortable became false when bug is modified.([code](https://github.com/spaceone-dev/spaceone-design-system/commit/20a80bfa78f50e97f8cb691a8aac4e2fc9284f47))
Due to the above issue, we have re-set rules for sorting.

Rules for props and fields
> * props.sortable: `true`  &&  field.sortable: `false`  =>  `false`
> * props.sortable: `true`  &&  field.sortable: `true`  =>  `true`
> * props.sortable: `false`  &&  field.sortable: `true`  =>  `false`
> * props.sortable: `false`  &&  field.sortable: `false`  =>  `false`
> * props.sortable: `true`  &&  field.sortable: `undefined`  =>  `true`
> * props.sortable: `false`  &&  field.sortable: `undefined`  =>  `false`

### CODE REVIEW

File: src/data-display/tables/data-table/PDataTable.stories.mdx [hunk 1]
```diff
@@ -222,6 +222,14 @@ This only affects the display of arrows based on the value of `sortDesc` and `so
...
 To sort data, receive the changed `sortBy` and `SortDesc` values through the `changeSort` or `update:sortBy` and `udpate:sortDesc` events, and use them in a way to reflect the changes in the `items` props. <br/>
 
+### Rules for props and fields
+> * props.sortable: `true`  &&  field.sortable: `false`  =>  `false`
+> * props.sortable: `true`  &&  field.sortable: `true`  =>  `true`
+> * props.sortable: `false`  &&  field.sortable: `true`  =>  `false`
+> * props.sortable: `false`  &&  field.sortable: `false`  =>  `false`
+> * props.sortable: `true`  &&  field.sortable: `undefined`  =>  `true`
+> * props.sortable: `false`  &&  field.sortable: `undefined`  =>  `false`
+
```
Comment (WANZARGEN) [inline src/data-display/tables/data-table/PDataTable.stories.mdx:232]: It's great to give detail description!

File: src/data-display/tables/data-table/PDataTable.vue [hunk 2]
```diff
@@ -35,7 +35,7 @@
                                         <span class="th-contents"
                                               :class="{
                                                   [field.textAlign || DATA_TABLE_CELL_TEXT_ALIGN.left]: true,
-                                                  'has-icon': sortable && field.sortable || colCopy,
+                                                  'has-icon': (sortable && isFieldSortable(field.sortable)) || colCopy,
```
Comment (WANZARGEN) [inline src/data-display/tables/data-table/PDataTable.vue:38]: How about changing the way to check sortable like this?

```suggestion
                                                  'has-icon': (isFieldSortable(field.sortable) || colCopy,
```

File: src/data-display/tables/data-table/PDataTable.vue [hunk 3]
```diff
@@ -169,11 +169,13 @@ interface TableField extends DataTableFieldType {
 
 const getChildFields = (field: DataTableFieldType): DataTableFieldType[]|undefined => field.children?.map(child => ({ sortable: true, ...child }));
 
+const isFieldSortable = (sortable: boolean | undefined): boolean => [undefined, true].includes(sortable);
```
Comment (WANZARGEN) [inline src/data-display/tables/data-table/PDataTable.vue:172]: How about changing `isFieldSortable` to check both sortable prop and field's sortable?

```suggestion
const isFieldSortable = (sortable: boolean | undefined): boolean => {
  if (props.sortable) {
    if (!soratable) return false;
    return true;
  }
  return false;
}
```

**Review by WANZARGEN — APPROVED** [review]
It is very good that the storybook has a detailed explanation of the sorting condition.

It would be better if the way of determining the sortable of each field could be further improved in a way that can improve readability.

File: src/data-display/tables/data-table/PDataTable.vue (hunk 3 above)
Comment (JeoungSulMo) [reply src/data-display/tables/data-table/PDataTable.vue:172]: isFieldSortable function is located outside of setup function. The isFieldSortable function and getTableField function need arguments for props.sortable. would it be okay?

Comment (WANZARGEN) [reply src/data-display/tables/data-table/PDataTable.vue]: yes! it would be okay!

**Pull request merged**